### PR TITLE
src/Makefile: add target to flash firmware for convenience

### PR DESCRIPTION
**Pull request by ia**
Hello. Just decided to share this tiny improvement for the main `Makefile`: in the case of _compile-flash-run_ cycle, now just putting the device into _dfu mode_ & running `make flash` will do the rest automagically. Can't wait to hear any feedback about this patch.

### CODE REVIEW

**Comment by bvernoux** [comment]
Thanks for the contribution, it is simple and shall do not have any border effect (I have not tested it so far)
The only potential issue is it is specific to Linux (as Windows support will requires a dedicated special version of dfu-util with same syntax)
@Baldanos could you check you do not see any issue on Linux ?

File: src/Makefile [hunk 1]
```diff
@@ -344,3 +344,12 @@ PRE_MAKE_ALL_RULE_HOOK: ./common/hydrafw_version.hdr
...
 POST_MAKE_ALL_RULE_HOOK: $(BUILDDIR)/$(PROJECT).dfu
+
+# Custom rule to flash firmware when hydrabus is connected in dfu mode
+flash: $(BUILDDIR)/$(PROJECT).dfu
+	@echo Flashing $^
+ifeq ($(OS), Windows_NT)
+	@../utils/windows_dfu_util/DfuSeCommand.exe -c --de 0 -d --fn $^
+else
+	@dfu-util -d 0483:df11 -a 0 -R -D $^
+endif
```
Comment (ia) [inline src/Makefile:355]: Here you go. Unfortunately, I can not test it on Windows myself but _it seems_ it should work something like that.

Comment (bvernoux) [reply src/Makefile:355]: I have tested `make flash` with MSYS2 and Windows11Pro and all work fine
Now testing on native Ubuntu22.04 LTS

Comment (bvernoux) [reply src/Makefile:355]: Thanks for your contribution

**Pull request merged**